### ISSUE TITLE
[Bug][jssrc] Methods from custom code appear in external methods list

### ISSUE THREAD

**Issue by prabhu** [joernio/joern#2734]
**Describe the bug**
Scan a project like [NodeGoat](https://github.com/OWASP/NodeGoat)

```
cpg.method.isExternal(true).fullName.l
```

Methods like below should be considered internal methods.

```
"app/assets/js/tour/redirects-steps.js::program:$:<returnValue>:on",
"app/assets/js/tour/redirects-steps.js::program:$",
"app/data/allocations-dao.js::program:callback",
"UserDAO:getUserById",
"ANY[]:push",
"__ecma.Array:push",
"{ userId: ANY; stocks: ANY; funds: ANY; bonds: ANY; }:forEach",
```

**To Reproduce**
1. Use NodeGoat

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS
 - Joern Version 1.1.1718
 - Java version 19

**Additional context**

**Comment by max-leuthaeuser** [comment]
Pretty sure those are coming from the type recovery mechanism @DavidBakerEffendi ?

**Comment by DavidBakerEffendi** [comment]
Yes @max-leuthaeuser. In summary, these method full names are built mostly using string concatenation as a last resort if the corresponding type/method cannot be correctly resolved. Now, this is most commonly the case for external dependencies/library code, so it will be considered an external method stub. The logic is basically that if the method is in the CPG, then it is internal, but if it cannot find it internally, then it is external.

It is a bit of a binary view, but if you are in doubt, then you can check the associated namespace, such methods are under `<speculatedMethods>` namespace.
